src/prompts/index.tsx

```diff
@@ -198,6 +204,55 @@ export class PromptManager {
         return false;
     }
 
+    private maybePersistReminder(prompt: PromptConfig) {
+        if (prompt.type !== 'alert') {
+            return;
+        }
+
+        const reminder = prompt.declinedPromptReminderDelay
+            ? { promptUuid: prompt.uuid, declinedOn: Date.now() }
+            : 'suppressed';
+
+        persistPromptReminder(
+            prompt.uuid,
+            reminder
+        );
+    }
+
+    private hidePrompt(prompt: PromptConfig) {
+        const idx = this.activePrompts.indexOf(prompt);
+        this.activePrompts.splice(idx, 1);
+
+        this.render();
+    }
+
+    private async isPromptSuppressed(prompt: PromptConfig): Promise<boolean> {
+        if (prompt.type !== 'alert') {
+            return Promise.resolve(false);
+        }
+
+        const reminder = await getPromptReminder(prompt.uuid);
+
+        if (!reminder) {
+            return false;
+        }
+
+        if ('suppressed' === reminder) {
```

Comment:
Just so I understand, this would make the default behaviour that if you don't configure any reminder delay, you'll never see the prompt again?

That's kind of the inverse of what I'd initially thought, but perhaps it makes most sense. Otherwise if you did want the prompt to never show again, you'd have to set a stupid long delay.

Reply:
Impl changed a little around this area with introduction to change to config which now explicitly determines the behaviour ('decline/suppress' or 'remind')

Reply:
Are we planning to extend this to consider the case where we don't want a delay/suppression for cases like any time the user completes an order, offer order updates prompt?

Reply:
Yeah, can just extend this with another action e.g ('Always show again') - if sounds good might iterate upon this when adding the additional prompt types which will be part of the next/current round of changes?

Reply:
I think we should have always show again as an option, but yeah can follow up in next iteration.